src/core/api.ts

```diff
@@ -86,19 +89,19 @@ class API {
    * @param {Object} data An object of data to post
    * @returns {CancelableRequest} request
    */
-  sendPostCancelable(url, args, opts, data, headers) {
-    let cancel;
+  sendPostCancelable(url: string, args?: string[], opts?: KeyValue, data?: any, headers?: KeyValue) {
+    const source: CancelTokenSource = axios.CancelToken.source()
     const conn = this.con()({
-      method: "post",
+      method: 'post',
       url,
       headers: createHeaders(args, opts, headers),
       data,
-      cancelToken: new CancelToken(function executor(c) {
-        // An executor function receives a cancel function as a parameter
-        cancel = c;
-      })
-    });
-    return { conn, cancel };
+      cancelToken: source.token
+    })
+    // TODO: fix cancel method return
+    // ISSUE: Cancel method above isn't set by time of return
+    // RETURN: { conn, cancel }
+    return { conn, source }
```

Comment:
Ideally here we'd return `source.cancel`, so that the user of the API simply has to call `cancel()` on the returned object. But there are other issues to figure out, so can change that later.

Reply:
yeah... method isn't in use yet so i just kept it straight forward

Reply:
👍